src/main/webapp/app/entities/search/search-modal.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useEffect, useState } from 'react';
+import {
+  View,
+  Flex,
+  Dialog,
+  Heading,
+  Divider,
+  Content,
+  Form,
+  ButtonGroup,
+  Button,
+  TextField,
+  TextArea,
+  AlertDialog,
+  DialogContainer,
+} from '@adobe/react-spectrum';
+import { IRootState } from 'app/shared/reducers';
+import { connect } from 'react-redux';
+import {translate, Translate} from 'react-jhipster';
+import { RouteComponentProps} from 'react-router-dom';
+import {resetSearch} from "app/entities/search/search.reducer";
+
+export interface ISearchModalProps extends StateProps, DispatchProps, RouteComponentProps<{ id: string, viewId: string }> {}
+
+const SearchModal = (props: ISearchModalProps) => {
+  const [searchText, setSearchText] = useState('')
+
+  useEffect(() => {
+    if (!props.isSearchOpen) {
+      props.history.push(`/dashboards/${props.match.params.id}/${props.match.params.viewId}/build`);
+    }
+  }, [props.isSearchOpen]);
+
+  const handleClose = () => {
+    props.resetSearch();
+  };
+
+  const handleSearchClick = () => {
+
+  };
+
+  return (
+    <>
+      <DialogContainer type="fullscreenTakeover"
+                       onDismiss={handleClose}>
+        <Dialog>
+          <Heading>
+            <Translate contentKey="views.search.title">_Search</Translate>
+          </Heading>
+          <Divider />
+          <Content>
+            <Flex direction="column" gap="size-100" alignItems="center">
+              <View padding="size-600">
+                <Form isRequired
```

Comment:
you will need to write a logic for isRequired

Reply:
this is just the UI, business logic is coming in next PRs